File: ts_version/src/agents/registry.ts
// TypeScript equivalent of agents/registry.py with enhanced registry patterns
import { Agent } from '../types';
import { benefitsAgent } from './benefits';
import { getGlobalAgentRegistry, AgentFactory } from './AgentRegistry';

// Available agent identifiers
export enum AvailableAgents {
  BENEFITS = 'benefits'
}

// Initialize the enhanced registry with available agents
const registry = getGlobalAgentRegistry();
registry.register(benefitsAgent);

// Create factory instance for advanced agent creation
export const agentFactory = new AgentFactory(registry);

/**
 * Get an agent instance by name
 * @param agentName - Name of the agent to retrieve
 * @returns Agent instance
 * @throws Error if agent not found
 */
export function getAgent(agentName: string): Agent {
  return registry.getRequiredAgent(agentName);
}

/**
 * Get multiple agent instances by names
 * @param agentNames - Array of agent names
 * @returns Array of agent instances
 * @throws Error if any agent not found
 */
export function getAgents(agentNames: string[]): Agent[] {
  return registry.getAgents(agentNames);
}

/**
 * Get all available agent names
 * @returns Array of available agent names
 */
export function getAvailableAgentNames(): string[] {
  return registry.getAgentNames();
}

/**
 * Check if an agent exists
 * @param agentName - Name of the agent to check
 * @returns True if agent exists
 */
export function hasAgent(agentName: string): boolean {
  return registry.hasAgent(agentName);
}

/**
 * Get agent metadata for all registered agents
 * @returns Array of agent metadata
 */
export function getAgentMetadata() {
  return registry.getAgentMetadata();
}

/**
 * Get the enhanced registry instance for advanced operations
 * @returns AgentRegistry instance
 */
export function getAgentRegistry() {
  return registry;
}

File: py_version/agents/registry.py
from enum import Enum
from typing import Dict
from schemas import Agent
from agents.benefits import benefits_agent


class AvailableAgents(Enum):
    BENEFITS = "benefits"


# Mapping of agent names to agent instances
AGENT_REGISTRY: Dict[str, Agent] = {
    AvailableAgents.BENEFITS.value: benefits_agent,
}


def get_agent(agent_name: str) -> Agent:
    """Get an agent instance by name"""
    if agent_name not in AGENT_REGISTRY:
        raise ValueError(f"Agent '{agent_name}' not found. Available agents: {list(AGENT_REGISTRY.keys())}")
    return AGENT_REGISTRY[agent_name]


def get_agents(agent_names: list[str]) -> list[Agent]:
    """Get multiple agent instances by names"""
    return [get_agent(name) for name in agent_names]